replica/database.cc

```diff
@@ -943,16 +951,23 @@ void database::maybe_init_schema_commitlog() {
 void database::add_column_family(keyspace& ks, schema_ptr schema, column_family::config cfg) {
     schema = local_schema_registry().learn(schema);
     schema->registry_entry()->mark_synced();
+    auto&& rs = ks.get_replication_strategy();
+    locator::effective_replication_map_ptr erm;
+    if (auto pt_rs = rs.maybe_as_per_table()) {
+        erm = pt_rs->make_replication_map(schema->id(), _shared_token_metadata.get());
+    } else {
+        erm = ks.get_effective_replication_map();
+    }
```

Comment:
I different approach is to change all call-sites to treat replication map creation as per-table, and adjust the creation code to hand a copy of previously created replication map if the replication strategy knows it's table-independent. It may reduce the churn.

Reply:
That would spread the logic to many places, right now it's in a single place.

Reply:
ok